lib/bot_framework/message.rb

```diff
@@ -0,0 +1,92 @@
+module BotFramework
+  class Message
+    TEXT_FORMAT = {
+      plain: 'plain',
+      markdown: 'markdown',
+      xml: 'xml'
+    }
+    ATTACHMENT_LAYOUT = {
+      list: 'list',
+      carousel: 'carousel'
+    }
+
+    def initialize(session = nil)
+      @data = {}
+      @data[:type] = 'consts.MessageType' #FIXME
+      @data[:agent] = 'consts.agent'
+      if @session
+        m = @session.message
+        @data[:source] = m[:source] if m[:source]
+        @data[:text_locale] = m[:text_locale] if m[:text_locale]
+        @data[:address] = m[:address] if m[:address]
+      end
+    end
+    
+    def text_locale(locale)
+      @data[:text_locale] = locale
+      self
+    end
+
+    def text_format(style)
+      @data[:text_format] = style
+      self
+    end
+
+    def text(text,*args)
+      @data[:text] = text.present? ? format_text(text) : ''
+      self
+    end
+
+    def ntext(msg,msg_plural,count)
```

Comment:
Unused method argument - `msg_plural`. If it's necessary, use `_` or `_msg_plural` as an argument name to indicate that it won't be used.